File: NodeJs/feature-functions/feature-functions.js
/* eslint-disable new-cap */
import { LexActivator, LexStatusCodes, LexActivatorException, PermissionFlags } from '@cryptlex/lexactivator'

function init() {
    /*
     * Product data needs to be hard-coded and this value is not sensitive or secret.
     * You can download the product.dat file from the product page of the admin dashboard.
     */
    LexActivator.SetProductData('PASTE_CONTENT_OF_PRODUCT.DAT_FILE');

    /*
     * Product Id is simply a unique identifier of your product and needs to be hard-coded.
     * Similarly this Id is not sensitive as well.
     */

    LexActivator.SetProductId('PASTE_PRODUCT_ID', PermissionFlags.LA_USER);
}

function licenseCallback(status) {
    try {
        switch (status) {
            case LexStatusCodes.LA_OK:
            case LexStatusCodes.LA_EXPIRED:
            case LexStatusCodes.LA_SUSPENDED:
                console.log('License is genuinely activated!', status);
                break;
            default:
                throw new LexActivatorException(status);
        }
    } catch (error) {
        console.log(error.code, error.message);
    }
}

function main() {
    try {
        init();
        /*
         * Whenever the server sync occurs in a separate thread, and server returns the response,
           license callback function gets invoked.
         */
        LexActivator.SetLicenseCallback(licenseCallback);

        const status = LexActivator.IsLicenseGenuine();
        if (LexStatusCodes.LA_OK == status) {
            console.log('License is genuinely activated!');

            // You can execute your code as the user is licensed

            const expiryDate = LexActivator.GetLicenseExpiryDate();
            const daysLeft = (expiryDate - (new Date().getTime() / 1000)) / 86400;
            console.log('Days left:', daysLeft);

            // Using License metadata for feature management

            // Changes made to the metadata value will get reflected after the server sync.

            const featureA = LexActivator.GetLicenseMetadata("MetadataKey");
            console.log("Metadata value: ", featureA.value);
            if (featureA.value == "allowed") {
                // user can use the feature
            } else {
                // what ever you want to do
            }

            // Changes made to the product version or feature flag will get reflected after the server sync.

            // Using cryptlex's feature management

            const featureObject = LexActivator.GetProductVersionFeatureFlag("featureName");
            console.log(featureObject.data)  // data that was passed with the feature
            console.log(featureObject.enabled) // whether the feature is enabled or not
            if (featureA.enabled) {
                // user can use the feature
            } else {
                // whatever you want to do
            }
        }
        else if (LexStatusCodes.LA_EXPIRED == status) {
            console.log('License is genuinely activated but has expired!');
        } else if (LexStatusCodes.LA_SUSPENDED == status) {
            console.log('License is genuinely activated but has been suspended!');
        } else if (LexStatusCodes.LA_GRACE_PERIOD_OVER == status) {
            console.log('License is genuinely activated but grace period is over!');
        }

    } catch (error) {
        console.log(error.code, error.message);
    }
}

main();

File: C-C++/feature-functions/feature-functions.cpp
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <string.h>
#include "../header/LexActivator.h"

using namespace std;

// Uncomment following for Windows static build
/*
#if _WIN32
#define LEXACTIVATOR_STATIC
#pragma comment(lib, "winhttp")
#if _WIN64
#pragma comment(lib, "x64/libcurl_MD")
#pragma comment(lib, "x64/LexActivator")
#else
#pragma comment(lib, "x86/libcurl_MD")
#pragma comment(lib, "x86/LexActivator")
#endif
#endif
*/

#if _WIN32
#if _WIN64
#pragma comment(lib, "x64/LexActivator")
#else
#pragma comment(lib, "x86/LexActivator")
#endif
#endif

void init()
{
    int status;

    /*
     * Product data needs to be hard-coded and this value is not sensitive or secret.
     */

#if _WIN32
    status = SetProductData(L"PASTE_CONTENT_OF_PRODUCT.DAT_FILE");
#else
    status = SetProductData("PASTE_CONTENT_OF_PRODUCT.DAT_FILE");
#endif
    if (LA_OK != status)
    {
        printf("Error code: %d", status);
        getchar();
        exit(status);
    }

    /*
     * Product Id is simply a unique identifier of your product and needs to be hard-coded.
     * Similarly this Id is not sensitive as well.
     */

#if _WIN32
    status = SetProductId(L"PASTE_PRODUCT_ID", LA_USER);
#else
    status = SetProductId("PASTE_PRODUCT_ID", LA_USER);
#endif
    if (LA_OK != status)
    {
        printf("Error code: %d", status);
        getchar();
        exit(status);
    }
}

int main()
{
    init();

    int status = IsLicenseGenuine();
    if (LA_OK == status)
    {
        printf("License is genuinely activated!");
        getchar();

        // You can now execute your code as the user is genuine/licensed.

        /*          #### Using License metadata for feature management ####         */

        // Changes made to the metadata value will get reflected after the server sync.

        char value[256];
#if _WIN32
        status = GetLicenseMetadata(L"FeatureA", value, 265);
#else
        status = GetLicenseMetadata("FeatureA", value, 256);
#endif
        if (LA_OK == status)
        {
            if (strcmp(value, "allowed") == 0)
            {
                // user can use this feature
            }
            else
            {
                printf("FeatureA not allowed");
            }
        }
        else
        {
            printf("Error code: %d", status);
            getchar();
            exit(status);
        }

        /*       #### Using cryptlex's feature management ####      */

        // Changes made to the product version or feature flag will get reflected after the server sync.

        unsigned int enabled; // it can either be true or false depicting if the feature is enabled or disabled
        wchar_t featureDataA[256];
        char featureData[256]; // data passed with the feature.

#if _WIN32
        status = GetProductVersionFeatureFlag(L"feature-1", &enabled, featureDataA, 256);
#else
        status = GetProductVersionFeatureFlag("feature-1", &enabled, featureData, 256);
#endif

        if (LA_OK == status)
        {
#if _WIN32
            printf("enabled:  %d\n", enabled);
            printf("Data: %c\n", featureDataA);
#else
            printf("enabled:  %d\n", enabled);
            printf("Data: %s\n", featureData);
#endif
        }
        else
        {
            printf("Error code: %d", status);
            getchar();
            exit(status);
        }
    }
    else if (LA_EXPIRED == status)
    {
        printf("License is genuinely activated but has expired!");
    }
    else if (LA_SUSPENDED == status)
    {
        printf("License is genuinely activated but has been suspended!");
    }
    else if (LA_GRACE_PERIOD_OVER == status)
    {
        printf("License is genuinely activated but grace period is over!");
    }
    else
    {
        printf("Something went wrong! Error code: %d", status);
    }
    getchar();
    return 0;
}